api/app/routers/hfi_calc.py

```diff
@@ -79,100 +93,121 @@ def extract_selected_stations(request: HFIResultRequest) -> List[int]:
+
+
 @router.post('/', response_model=HFIResultResponse)
 async def get_hfi_results(request: HFIResultRequest,
                           response: Response,
                           token=Depends(authentication_required)):
     """ Returns calculated HFI results for the supplied details in the POST body """
+    # Yes. There are a lot of locals!
+    # pylint: disable=too-many-locals
 
     try:
         logger.info('/hfi-calc/')
         response.headers["Cache-Control"] = no_cache
 
-        request = load_request(request)
+        stored_request = load_request_from_database(request)
+        request_loaded = False
+        if stored_request:
+            request = stored_request
+            request_loaded = True
 
         # ensure we have valid start and end dates
         valid_start_date, valid_end_date = validate_date_range(request.start_date, request.end_date)
         # wf1 talks in terms of timestamps, so we convert the dates to the correct timestamps.
         start_timestamp = int(app.utils.time.get_hour_20_from_date(valid_start_date).timestamp() * 1000)
         end_timestamp = int(app.utils.time.get_hour_20_from_date(valid_end_date).timestamp() * 1000)
 
-        selected_prep_date = request.selected_prep_date
-        if selected_prep_date is None \
-                or selected_prep_date < valid_start_date or selected_prep_date > valid_end_date:
-            selected_prep_date = valid_start_date
-
         async with ClientSession() as session:
             header = await get_auth_header(session)
             # TODO: Enable when fuel type config implemented
             # selected_station_codes = extract_selected_stations(request)
-            wfwx_stations = await app.wildfire_one.wfwx_api.get_wfwx_stations_from_station_codes(
-                session, header, request.selected_station_code_ids)
-            dailies = await get_dailies_lookup_fuel_types(
-                session, header, wfwx_stations, start_timestamp, end_timestamp)
-            prep_delta = valid_end_date - valid_start_date  # num prep days is inclusive
-            # NOTE: database session brought to this level in order to make code review of
-            # calculate_hfi_results easier. (adding session in there, results in the entire function
-            # being indented, which makes code review difficult.) Please move session back into
-            # function in isolated pr.
+
             with get_read_session_scope() as orm_session:
-                results = calculate_hfi_results(request.selected_fire_center_id,
-                                                request.planning_area_fire_starts,
-                                                dailies, prep_delta.days,
-                                                request.selected_station_code_ids,
-                                                orm_session)
+                # Fetching dailies is an expensive operation. When a user is clicking an unclicking stations
```

Comment:
Similar to request stuff, this could be an independent class/module that handles the glueing up and caching of data here.

Reply:
I'll do that in the next PR - I'll move it to app.hfi (already too many things moving around in this PR!)